### PR TITLE
fix(accordion): fixing header tag, fixing heading text attribute  

### PR DESCRIPTION
**Pull request by brianferry**
## What I did

1. Added an `anonymous` slot checker in the `pf-accordion-header` component. 
  - This checks for whether an `anonymous` slot has been passed in for the header
    - DOES NOT have an `anonymous` slot = we add the header tag to the shadow dom
    - DOES have an `anonymous` slot = we DO NOT add the header tag to the shadow dom
 2. We have / had an issue with the `heading-text` attribute not showing correctly in the header, this has been fixed
 3. Adding a demo to show the attributes being used instead of the `anonymous` slot

## Testing Instructions

1. [DP](https://deploy-preview-2435--patternfly-elements.netlify.app/components/accordion/demo/)
2. Open the developer console and inspect the accordion demos
3. Verify that the accordion headers with slotted in content do not have nested header tags in the shadow dom
4. Verify that the accrdion headers in the demo that have text / tags passed in via attributes have the nested header tags in the shadow dom. 
5. Verify that the text passed in through `heading-text` is showing in the header as expected

### CODE REVIEW

File: elements/pf-accordion/BaseAccordionHeader.ts [hunk 1]
```diff
@@ -71,10 +79,23 @@ export abstract class BaseAccordionHeader extends LitElement {
   renderAfterButton?(): TemplateResult;
 
   override render(): TemplateResult {
+    const hasAnonymousSlot = this.#slots.getSlotted()?.length > 0;
     const tag = unsafeStatic(this.headingTag);
     const ariaExpandedState = String(!!this.expanded) as 'true'|'false';
     return staticH`
-      <${tag} id="heading">
+      ${!hasAnonymousSlot ?
+        staticH`
```
Comment (bennypowers) [inline elements/pf-accordion/BaseAccordionHeader.ts:87]: see https://github.com/RedHat-UX/red-hat-design-system/pull/787

**Review by bennypowers — CHANGES_REQUESTED** [review]

File: elements/pf-accordion/BaseAccordionHeader.ts
Comment (bennypowers) [inline elements/pf-accordion/BaseAccordionHeader.ts:1]: @brianferry please review this last refactoring commit and if it's good, resolve this comment to merge

**Review by bennypowers — APPROVED** [review]

File: elements/pf-accordion/BaseAccordionHeader.ts
Comment (bennypowers) [reply elements/pf-accordion/BaseAccordionHeader.ts:1]: also, please add a changeset which describes the user-facing changes

Comment (bennypowers) [reply elements/pf-accordion/BaseAccordionHeader.ts:1]: this should also be tested against rh-accordion

Comment (brianferry) [reply elements/pf-accordion/BaseAccordionHeader.ts:1]: @heyMP - 

image

Pulled in the new updated pf-accordion files & built & ran it w/ `main` for footer, no longer seeing the shadow-root h3s, wanting to confirm everything looks good to you too

**Review by bennypowers — APPROVED** [review]

**Pull request merged**